sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/taskdata/AuxLabelAccess.java

```diff
@@ -32,16 +32,39 @@ private AuxLabelAccess() {
         // do not instantiate
     }
 
-    // Resource ID
+    // Reservation labels: Resource ID and Service Name
 
-    public static void setResourceId(Protos.Resource.ReservationInfo.Builder reservationBuilder, String resourceId) {
-        reservationBuilder.setLabels(
-                withLabel(reservationBuilder.getLabels(), LabelConstants.RESOURCE_ID_RESERVATION_LABEL, resourceId));
+    /**
+     * Assigns labels to the provided reservation:
+     * <ul><li>The service name which owns the reservation</li>
+     * <li>A UUID for tracking the reservation</li></ul>
+     *
+     * @param reservationBuilder the reservation where labels should be added
+     * @param serviceName the name of the service which owns the reservation
+     * @param resourceId a unique id for tracking the reservation
+     */
+    public static void setResourceLabels(
+            Protos.Resource.ReservationInfo.Builder reservationBuilder, String serviceName, String resourceId) {
+        Map<String, String> map = LabelUtils.toMap(reservationBuilder.getLabels());
+        map.put(LabelConstants.RESOURCE_ID_RESERVATION_LABEL, resourceId);
+        map.put(LabelConstants.SERVICE_NAME_RESERVATION_LABEL, serviceName);
```

Comment:
Does this allow slashes and we don't need to sanitize it?

Reply:
I assume yes

Reply:
Yes. These are arbitrary data as far as Mesos is concerned.

Looking at sanitizing service name in TaskIDs now. Shouldn't be a big deal, just hadn't tried it during initial development. Had a TODO somewhere to check on it...